build/dockerfiles/rhel.install.sh

```diff
@@ -8,7 +8,20 @@
 # SPDX-License-Identifier: EPL-2.0
 #
 
-microdnf install -y findutils bash wget yum gzip tar jq python3-six python3-pip skopeo && microdnf -y clean all && \
+architecture=""
+case $(uname -m) in
+       x86_64) architecture="amd64" ;;
+       s390x) architecture="s390x";;
+esac
+if [ $architecture == "amd64" ]; then \
+       microdnf install -y findutils bash wget yum gzip tar jq python3-six python3-pip skopeo && microdnf -y clean all; \
+elif [ $architecture == "s390x" ]; then \
+       microdnf install -y git findutils bash wget yum gzip tar python3-six python3-pip && microdnf -y clean all && \
+       for i in g/glibc-common-2.29-9 g/glibc-all-langpacks-2.29-9 g/glibc-2.29-9 o/oniguruma-6.9.1-2 j/jq-1.6-2 s/skopeo-0.1.36-6.dev.gitd93a581 c/containers-common-0.1.36-6.dev.gitd93a581; do wget https://rpmfind.net/linux/fedora-secondary/releases/30/Everything/s390x/os/Packages/$i.fc30.s390x.rpm; done; \
```

Comment:
hardcoding the versions here is super brittle and means you'll have to manually update to newer versions to fix CVE issues. Can you fetch rpms w/o needing to use specific versions?

Reply:
also it's recommended that you use a version of skopeo >= 0.1.40 so that you have access to the new `skopeo copy --all` command that allows copying multiple images/arches.

You might not actually NEED to do that here, but might as well use a current/recent version so containers and tools are on the same playing field.

Reply:
@nickboldt We did further investigation on your review comments and found that Fedora rpm's till version 30 only can be installed on ubi8 distribution. The later releases of Fedora have zstd compression support in their rpms which ubi8 do not support currently. Thus we cannot install latest Fedora rpms in ubi8 container. Ref. https://bugzilla.redhat.com/show_bug.cgi?id=1715799
Kindly let us know if you have any workaround in your knowledge, else we will have to fallback to 'install from source' strategy.

Reply:
That's been my experience too... up to and including F30, GOOD. F31+, can't be installed on RHEL8 (at least not on 8.1). 8.2 might add Zstd compression (I don't know, maybe).

Here's how I install Skopeo 0.1.40 from F30 in RHEL7:

```
   yum -y install podman-docker skopeo

    # CRW-563 install skopeo 0.1.40 from Fedora 30 repo
    # NOTE: DO NOT USE Fedora 31 -uses Zstd compression not avail in RHEL 7.7
    # >>> yum error: rpmlib(PayloadIsZstd) <= 5.4.18-1 is needed by skopeo-1:0.1.40-4.fc31.x86_64

    yum-config-manager --add-repo http://download.fedoraproject.org/pub/fedora/linux/updates/30/Everything/x86_64 || true
    yum update -y -q skopeo containers-common --nogpgcheck  --disablerepo="*" \
        --enablerepo=download.fedoraproject.org_pub_fedora_linux_updates_30_Everything_x86_64
    rm -fr /etc/yum.repos.d/download.fedoraproject.org_pub_fedora_linux_*
    skopeo --version
```

Reply:
is there a way to have same instructions between all arch ?

Reply:
we could enable Fedora 30 repo for `$MY_ARCH` in both cases, yes.

```
   MY_ARCH="$(uname -m)" &&
   if [[ ${MY_ARCH} == "x86_64" ]]; then export MY_ARCH="amd64"; fi && \

    yum-config-manager --add-repo http://download.fedoraproject.org/pub/fedora/linux/updates/30/Everything/${MY_ARCH} || true
    yum update -y -q skopeo containers-common --nogpgcheck  --disablerepo="*" \
        --enablerepo=download.fedoraproject.org_pub_fedora_linux_updates_30_Everything_${MY_ARCH}
```

Reply:
in the above example we would also install/update `yq`, etc.

Reply:
@nickboldt  We have tried above code given by you, but have observed few things as follows:
1. The repository that you mentioned in the code 'http://download.fedoraproject.org/pub/fedora/linux/updates/30/Everything/ does not have directory for s390x hence facing 404 error.
2. Also faced yum-config-manager:command not found error, hence tried installing it using yum-utils but in ubi8 yum command is not available in ubi-minimal containers.

Hence We use the following approach as workaround:
```echo '[fedora]
name=Fedora $releasever - $basearch
baseurl=https://dl.fedoraproject.org/pub/fedora-secondary/releases/30/Everything/s390x/os/
enabled=1
gpgcheck=0' > /etc/yum.repos.d/fedora.repo
       microdnf install -y glibc-common  glibc-all-langpacks glibc oniguruma containers-common skopeo jq && microdnf update -y skopeo containers-common jq oniguruma
```
The above code is working and has following major changes:
1. Changed repo to 'https://dl.fedoraproject.org/pub/fedora-secondary/releases/30/Everything/s390x/os/' which has all the dependencies and is stable too.

Can we use the above approach for all arch?
 WDYT?

Reply:
Yes, I don't see why you can't use https://dl.fedoraproject.org/pub/fedora-secondary/releases/30/Everything/ for s390x / ppc64le and http://download.fedoraproject.org/pub/fedora/linux/updates/30/Everything/ for x86_64